plugins/bulk-import/src/api/BulkImportBackendClient.ts

```diff
@@ -0,0 +1,192 @@
+  getImportJobs: (page: number, size: number) => Promise<ImportJobStatus[]>;
+  createImportJobs: (
+    importRepositories: CreateImportJobRepository[],
+    dryRun?: boolean,
+  ) => Promise<ImportJobResponse[]>;
+  checkImportStatus: (
+    repo: string,
+    defaultBranch: string,
+  ) => Promise<ImportJobStatus>;
+  removeRepository: (repo: string, defaultBranch: string) => Promise<Response>;
+};
+
+export type Options = {
+  configApi: ConfigApi;
+  identityApi: IdentityApi;
+};
+
+// @public
+export const bulkImportApiRef = createApiRef<BulkImportAPI>({
+  id: 'plugin.bulk-import.service',
+});
+
+export class BulkImportBackendClient implements BulkImportAPI {
+  // @ts-ignore
+  private readonly configApi: ConfigApi;
+  // private readonly identityApi: IdentityApi;
+
+  constructor(options: Options) {
+    this.configApi = options.configApi;
+    // this.identityApi = options.identityApi;
+  }
+
+  async getRepositories(page: number, size: number) {
+    // const { token: idToken } = await this.identityApi.getCredentials();
+
+    const backendUrl = this.configApi.getString('backend.baseUrl');
+    const jsonResponse = await fetch(
+      `${backendUrl}/api/bulk-import-backend/repositories?pagePerIntegration=${page}&sizePerIntegration=${size}`,
+      {
+        headers: {
+          // ...(idToken && { Authorization: `Bearer ${idToken}` }),
+          'Content-Type': 'application/json',
+        },
+      },
+    );
+    if (jsonResponse.status !== 200 && jsonResponse.status !== 204) {
+      return [];
+    }
+    return jsonResponse.json();
+  }
+
+  async getOrganizations(page: number, size: number) {
+    // const { token: idToken } = await this.identityApi.getCredentials();
+
+    const backendUrl = this.configApi.getString('backend.baseUrl');
+    const jsonResponse = await fetch(
+      `${backendUrl}/api/bulk-import-backend/organizations?pagePerIntegration=${page}&sizePerIntegration=${size}`,
+      {
+        headers: {
+          // ...(idToken && { Authorization: `Bearer ${idToken}` }),
+          'Content-Type': 'application/json',
+        },
+      },
+    );
+    if (jsonResponse.status !== 200 && jsonResponse.status !== 204) {
+      return [];
+    }
+    return jsonResponse.json();
+  }
+
+  async getRepositoriesFromOrg(orgName: string, page: number, size: number) {
+    // const { token: idToken } = await this.identityApi.getCredentials();
+
+    const backendUrl = this.configApi.getString('backend.baseUrl');
+    const jsonResponse = await fetch(
+      `${backendUrl}/api/bulk-import-backend/organizations/${orgName}/repositories?pagePerIntegration=${page}&sizePerIntegration=${size}`,
+      {
+        headers: {
+          // ...(idToken && { Authorization: `Bearer ${idToken}` }),
+          'Content-Type': 'application/json',
+        },
+      },
+    );
+    if (jsonResponse.status !== 200 && jsonResponse.status !== 204) {
+      return [];
+    }
+    return jsonResponse.json();
+  }
```

Comment:
This code block seems repeated , can be abstracted out into common fetcher

Reply:
created a common dataFetcher